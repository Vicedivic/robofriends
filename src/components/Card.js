import React from "react";

const Card = ({ name, email, id }) => {
  let image = null;
  try {
    image = require(`../img/${id}.jpg`);
  } catch (err) {
    // Image does not exist.
  }

  return (
    <div className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5">
      <img src={image} alt="robots" />
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
};

export default Card;
