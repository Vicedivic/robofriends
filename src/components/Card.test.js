import React from "react";
import ShallowRenderer from "react-shallow-renderer";
import { shallow } from "enzyme";
import Card from "./Card";

describe("Card", () => {
  const props = { id: 1, name: "John Doe", email: "john@doe.com" };
  const renderer = new ShallowRenderer();
  const cardHtml = renderer.render(<Card {...props} />);
  const card = shallow(<Card {...props} />);

  it("renders properly", () => {
    expect(cardHtml).toMatchSnapshot();
  });
});