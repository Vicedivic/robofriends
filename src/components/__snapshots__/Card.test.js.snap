// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders properly 1`] = `
<div
  className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
>
  <img
    alt="robots"
    src="1.jpg"
  />
  <div>
    <h2>
      John Doe
    </h2>
    <p>
      john@doe.com
    </p>
  </div>
</div>
`;
